File: Golf.Tournament/Models/Player.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace Golf.Tournament.Models
{
    public class Player
    {
        public Player()
        {
            Address = new Address();
            HomeClub = new Club();
        }

        [JsonProperty("_id")]
        public string Id { get; set; }

        [JsonProperty("firstname")]
        [Required]
        public string Firstname { get; set; }

        [JsonProperty("lastname")]
        [Required]
        public string Lastname { get; set; }

        [JsonProperty("address")]
        [Required]
        public Address Address { get; set; }

        [JsonProperty("homeClub")]
        public Club HomeClub { get; set; }
        

        [JsonProperty("hcp")]
        [Required]
        public decimal Hcp { get; set; }
    }
}

File: Golf.js/schemas/player.js

var playerSchema = {
    "id": "playerSchema", //important thing not to forget
    "$schema": "http://json-schema.org/draft-04/schema#",
    "title": "Player Schema",
    "type": "object",
    "properties": {
        "_id": {
            "type": ["string", "null", "object"]
        },
        "firstname": {
            "type": "string"
        },
        "lastname": {
            "type": "string"
        },
        "address": {
            "$ref": "addressSchema"
        },
        "homeClub": {
            "title": "Home Club Schema",
            "type": "object",
            "properties": {
                "name": {
                    "type": ["string", "null"]
                },
                "_id": {
                    "type": "string"
                }
            },
            "required": ["_id"]
        },
        "hcp": {
            "type": ["number", "integer"],
            "minimum": 0,
            "maximum": 54
        },
        "overrideHcp": {
            "type": ["null","boolean"]
        },
        "avatar": {
            "type": ["null", "string"]
        }
    },
    "required": ["firstname", "lastname", "homeClub", "hcp"]
}


module.exports = playerSchema;